Doc/library/timeit.rst

```diff
@@ -46,8 +46,17 @@ This can be achieved from the :ref:`python-interface` with::
 
 
 Note however that :mod:`timeit` will automatically determine the number of
-repetitions only when the command-line interface is used.  In the
-:ref:`timeit-examples` section you can find more advanced examples.
+repetitions only when the command-line interface is used.
+
+:mod:`timeit` also takes a callable for the statement and setup parameters::
+
+   >>> timeit.timeit(lambda: "-".join(map(str, range(100))), number=10000)
```

Comment:
This behavior doesn't seem to be tested in https://github.com/python/cpython/blob/master/Lib/test/test_timeit.py It would be nice to add some tests, so we won't break it accidentally in the future.

Reply:
@berkerpeksag isn't it tested? But what is `test_timeit_callable_stmt` doing then?

Reply:
@serhiy-storchaka sure, but that wasn't the point of that example?

Reply:
It is a bit silly to discuss complicated examples if there is a simpler way.

You need to use more complex ways only when you can't just pass a callable. For example when you need to pass arguments to the tested function.

Simpler way should be documented first, and more complex way should be shown for cases that can't be solved in simpler way.

Reply:
I guess... but it's also hard to come up with complicated examples that are also meaningful. Ok, but how about this:

```python
   def test(x):
       """Stupid test function"""
       L = [i for i in range(x)]

   num_iterations = 100

   if __name__ == '__main__':
       import timeit
       print(timeit.timeit("test(num_iterations)", setup="from __main__ import *"))
```

Reply:
ping?